connect/mirror-client/src/main/java/org/apache/kafka/connect/mirror/MirrorMakerConfig.java

```diff
@@ -0,0 +1,248 @@
+            Type.STRING,
+            CommonClientConfigs.DEFAULT_SECURITY_PROTOCOL,
+            Importance.MEDIUM,
+            CommonClientConfigs.SECURITY_PROTOCOL_DOC)
+        .withClientSslSupport()
+        .withClientSaslSupport();
+
+    static final ConfigDef ENABLED_CONFIG_DEF = new ConfigDef()
+        .define(ENABLED_CONFIG,
+            Type.BOOLEAN,
+            false,
+            Importance.HIGH,
+            ENABLED_DOC);
+
+    public MirrorMakerConfig(Map<?, ?> props) {
+        super(CONFIG_DEF, props, false);
+    }
+
+    public List<String> clusters() {
+        return getList(CLUSTERS_CONFIG);
+    }
+
+    public List<SourceAndTarget> enabledClusterPairs() {
+        return clusterPairs().stream()
+            .filter(this::enabled)
+            .collect(Collectors.toList());
+    }
+
+    public List<SourceAndTarget> clusterPairs() {
+        List<SourceAndTarget> pairs = new ArrayList<>();
+        List<String> clusters = clusters();
+        for (String source : clusters) {
+            for (String target : clusters) {
+                SourceAndTarget sourceAndTarget = new SourceAndTarget(source, target);
+                if (!source.equals(target)) {
+                    pairs.add(sourceAndTarget);
+                }
+            }
+        }
+        return pairs;
+    }
+
+    boolean enabled(SourceAndTarget sourceAndTarget) {
+        return new AbstractConfig(ENABLED_CONFIG_DEF, originalsWithPrefix(
+            sourceAndTarget.source() + "->" + sourceAndTarget.target() + "."), false)
+            .getBoolean(ENABLED_CONFIG);
+    }
+
+    /** Construct a MirrorClientConfig from properties of the form cluster.x.y.z.
+      * Use to connect to a cluster based on the MirrorMaker top-level config file.
+      */
+    public MirrorClientConfig clientConfig(String cluster) {
+        return new MirrorClientConfig(clusterProps(cluster));
+    }
+
+    // loads properties of the form cluster.x.y.z
+    Map<String, String> clusterProps(String cluster) {
+        Map<String, Object> props = new HashMap<>();
+
+        // fill in consumer, producer, admin configs
+        props.putAll(sharedClientConfigs());
+
+        // override with cluster-level properties
+        props.putAll(originalsWithPrefix(cluster + "."));
+
+        return toStrings(props);
+    }
+
+    // loads worker configs based on properties of the form cluster.x.y.z 
+    Map<String, String> workerConfig(SourceAndTarget sourceAndTarget) {
+        Map<String, String> props = clusterProps(sourceAndTarget.target());
+
+        if (props.get(GROUP_ID_CONFIG) == null) {
+            props.put(GROUP_ID_CONFIG, sourceAndTarget.source() + "-mm2");
+        }
+        if (props.get(OFFSET_STORAGE_TOPIC_CONFIG) == null) {
+            props.put(OFFSET_STORAGE_TOPIC_CONFIG, "mm2-offsets."
+                + sourceAndTarget.source() + ".internal");
+        }
+        if (props.get(STATUS_STORAGE_TOPIC_CONFIG) == null) {
```

Comment:
Usually the kafka convention of internal topic notation is double underscore, such as `__topic-name`. I wonder if we'd should to apply this here too. Also it seems to me that there internal topics are `mm2-offsets...`, `mm2-status...`, `mm2-offset-syncs` and `mm2-configs...` for the source and target clusters so we might be able just dynamically populate the topic blacklist for these and that way we'd probably leave the `.internal` notation.
Just putting this out for conversation.

Reply:
I think the `__` should be limited to topics internal to Kafka proper. The topics `mm2-offsets...`, `mm2-status...`, and `mm2-config...` are based on Connect's defaults, `connect-offsets`, `connect-config` etc, which are not internal per se. I've added .internal to these so that you don't need to configure MM2 to blacklist itself, thought it's an interesting idea to just blacklist them automatically.